tests/all_good.rs

```diff
@@ -0,0 +1,45 @@
+use test_harness::test;
+use test_harness::BaseMatcher;
+use test_harness::GhciWatchBuilder;
+use test_harness::Matcher;
+
+/// Test that `ghciwatch` can detect when compilation fails.
+///
+/// Regression test for DUX-1649.
```

Comment:
Perhaps create a matching GitHub issue that can be referenced here since this is an open source project (since outside contributors won't have access to our internal Linear).